packages/manager/modules/pci/src/projects/project/instances/instances.service.js

```diff
@@ -321,45 +323,48 @@ export default class PciProjectInstanceService {
     }).$promise;
   }
 
-  getSnapshotMonthlyPrice(projectId, instance, catalogEndpoint) {
-    return this.CucPriceHelper.getPrices(projectId, catalogEndpoint).then(
-      (catalog) => {
-        if (instance.planCode?.includes('3AZ')) {
-          return this.getProductAvailability(projectId).then(({ plans }) => {
-            const plan = plans.find(
-              ({ code, regions }) =>
-                code.startsWith('snapshot.consumption') &&
-                regions.find(({ name }) => name === instance.region),
-            );
-
-            return catalog[plan?.code] ?? catalog['snapshot.consumption.3AZ'];
-          });
-        }
-        if (instance.isLocalZone) {
-          return this.getProductAvailability(projectId).then(
-            ({ plans }) =>
-              catalog[
-                plans.find(
-                  ({ code, regions }) =>
-                    code.startsWith('snapshot.consumption') &&
-                    regions.find(({ name }) => name === instance.region),
-                )?.code
-              ] ?? catalog['snapshot.consumption.LZ'],
-          );
-        }
+  getSnapshotPricesByRegion(projectId, catalogEndpoint) {
+    return this.$q
+      .all([
+        this.CucPriceHelper.getPrices(projectId, catalogEndpoint, true),
+        this.getProductAvailability(projectId, undefined, 'snapshot'),
+      ])
+      .then(([catalog, { plans }]) => {
+        return plans
+          .filter(({ code }) => code.startsWith('snapshot.consumption'))
```

Comment:
nitpick: you could filter inside the reduce

Reply:
Same as above, I prefer to keep calls atomics